akka-actor/src/main/scala/akka/pattern/internal/CircuitBreakerTelemetry.scala

```diff
@@ -0,0 +1,157 @@
+ *
+ * {{{
+ * akka.circuit-breaker.telemetry.implementations += com.example.MyMetrics
+ * }}}
+ */
+@InternalStableApi
+trait CircuitBreakerTelemetry {
+
+  /**
+   * Invoked when the circuit breaker transitions to the open state.
+   */
+  def onOpen(): Unit
+
+  /**
+   * Invoked when the circuit breaker transitions to the close state.
+   */
+  def onClose(): Unit
+
+  /**
+   * Invoked when the circuit breaker transitions to the half-open state after reset timeout.
+   */
+  def onHalfOpen(): Unit
+
+  /**
+   * Invoked for each successful call.
+   *
+   * @param elapsedNanos the elapsed duration of the call in nanoseconds
+   */
+  def onCallSuccess(elapsedNanos: Long): Unit
+
+  /**
+   * Invoked for each call when the future is completed with exception, except for
+   * [[scala.concurrent.TimeoutException]] and [[akka.pattern.CircuitBreakerOpenException]]
+   * that are handled by separate methods.
+   *
+   * @param elapsedNanos the elapsed duration of the call in nanoseconds
+   */
+  def onCallFailure(elapsedNanos: Long): Unit
+
+  /**
+   * Invoked for each call when the future is completed with `java.util.concurrent.TimeoutException`
+   *
+   * @param elapsedNanos the elapsed duration of the call in nanoseconds
+   */
+  def onCallTimeoutFailure(elapsedNanos: Long): Unit
+
+  /**
+   * Invoked for each call when the future is completed with
+   * `akka.pattern.CircuitBreakerOpenException`
+   */
+  def onCallBreakerOpenFailure(): Unit
+
+  /**
+   * Called when the circuit breaker is removed, e.g. expired due to inactivity. It is also called
+   * if the circuit breaker is re-configured, before calling [[CircuitBreakerTelemetryProvider#start]].
+   */
+  def stopped(): Unit
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object CircuitBreakerTelemetryProvider {
+  def start(breakerId: String, system: ExtendedActorSystem): CircuitBreakerTelemetry = {
+    val configPath = "akka.circuit-breaker.telemetry.implementations"
+    if (!system.settings.config.hasPath(configPath)) {
+      CircuitBreakerNoopTelemetry
+    } else {
+      val telemetryFqcns: JList[String] = system.settings.config.getStringList(configPath)
+
+      telemetryFqcns.size() match {
+        case 0 =>
+          CircuitBreakerNoopTelemetry
+        case 1 =>
+          val fqcn = telemetryFqcns.get(0)
+          create(breakerId, system, fqcn)
+        case _ =>
+          new CircuitBreakerEnsembleTelemetry(telemetryFqcns.asScala.toSeq, breakerId, system)
+      }
+    }
+  }
+
+  def create(breakerId: String, system: ExtendedActorSystem, fqcn: String): CircuitBreakerTelemetry = {
+    system.dynamicAccess
+      .createInstanceFor[CircuitBreakerTelemetry](
+        fqcn,
+        List(classOf[String] -> breakerId, classOf[ExtendedActorSystem] -> system))
+      .get
+  }
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object CircuitBreakerNoopTelemetry extends CircuitBreakerTelemetry {
+  override def onOpen(): Unit = {}
```

Comment:
for empty methods returning Unit we typically use `= ()`

Reply:
Done ✅